### PR TITLE
Fail startup/network timeout earlier. Exit whole test suite.

### PR DESCRIPTION
**Pull request by selfhoster1312**
This greatly simplifies the VM starting logic:

- only try to start the VM once... why would it work after a failure when repeating the same steps?
- apply a hard 10s timeout (with SIGKILL) on attempts to check if systemd has booted and the network is available ; this means the initial setup step should at worst fail after ~100s (5x10 + 5x10)
- fail this entire `_LXC_CREATE` initial step and therefore the entire test suite if bringing up the VM failed... as the message used to say, it failed miserably, and so there is no reason to keep going and run into weirder and weirder errors such as the instance name not existing because it was not created

Works for me™

### CODE REVIEW

**Comment by selfhoster1312** [comment]
So it turns out those timeouts are too low in real life conditions... I don't understand why/how LXC can be so slow on a recent machine?!

File: lib/lxc.sh [hunk 1]
```diff
@@ -52,7 +52,13 @@ LXC_CREATE () {
         log_critical "Failed to create the new LXC :/"
     fi
 
-    _LXC_START_AND_WAIT $LXC_NAME
+    # The first time around when we create the VM, if it fails, we want to abort early
+    # instead of keeping the timeout game until the end of times
+    if ! _LXC_START_AND_WAIT $LXC_NAME; then
+        log_error "Fatal error creating VM. See logs above."
+        exit 1
```
Comment (alexAubin) [inline lib/lxc.sh:59]:
```suggestion
        log_critical "Fatal error creating container. See logs above."
```

File: lib/lxc.sh [hunk 2]
```diff
@@ -215,70 +221,42 @@ LXC_RESET () {
...
-		fi
-
-		# Fail if the container failed to start
-		if [ $i -eq $max_try ] && [ $failstart -eq 1 ]
-		then
-            log_error "The container miserably failed to start or to connect to the internet"
+    # Wait for container to start, we are using systemd to check this,
+    # for the sake of brevity.
+    for j in $(seq 1 5); do
+        log_debug "Start VM attempt $j"
```
Comment (alexAubin) [inline lib/lxc.sh:229]:
```suggestion
        log_debug "Start container attempt $j"
```

File: lib/lxc.sh [hunk 3]
```diff
@@ -215,70 +221,42 @@ LXC_RESET () {
...
+            return 1
+        fi
+    done
 
-    sleep 3
+    log_info "Container started successfully. Now waiting for internet access."
+
+    # Wait for container to access the internet
+    for j in $(seq 1 5); do
+        log_debug "Connect VM internet attempt $j"
```
Comment (alexAubin) [inline lib/lxc.sh:245]:
```suggestion
        log_debug "Connect container internet attempt $j"
```

**Pull request merged**